packages/user/src/supertokens/recipes/config/third-party-email-password/emailPasswordSignUp.ts

```diff
@@ -17,18 +17,10 @@ const emailPasswordSignUp = (
   const { config, log, slonik } = fastify;
 
   return async (input) => {
-    if (config.user.features?.signUp === false) {
-      throw {
-        name: "SIGN_UP_DISABLED",
-        message: "SignUp feature is currently disabled",
-        statusCode: 404,
-      } as FastifyError;
-    }
+    const roles = (input.userContext.roles || []) as string[];
 
-    const role = config.user.role || "USER";
-
-    if (!(await isRoleExists(role))) {
-      log.error(`Role "${role}" does not exist`);
+    if (!(await areRolesExist(roles))) {
+      log.error(`Roles ${roles} does not exist`);
```

Comment:
I think we need to change this message, this message implies all the roles from context is not exist. instead we have to log which role is not exists in the system or need to set the error something like one or more roles are not exists.